src/self_driving/planning/pilot_util.cpp

```diff
@@ -27,111 +28,76 @@
+      bool success = true;
+      size_t idx = util.CompileQuery(sql_query, nullptr, nullptr, &success);
+      if (success) util.ExecuteQuery(idx, nullptr, nullptr, nullptr);
+      std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
   } else {
-    // For all other actions, we apply the action by running it as query in a committed transaction
-    txn = txn_manager->BeginTransaction();
-
-    auto accessor = catalog->GetAccessor(common::ManagedPointer(txn), db_oid, DISABLED);
-
     // Parameters are also specified in the query string, hence we have no parameters nor parameter types here
-    auto out_plan =
-        PilotUtil::GenerateQueryPlan(txn, common::ManagedPointer(accessor), nullptr, nullptr, statement->ParseResult(),
-                                     db_oid, pilot->stats_storage_, pilot->forecast_->GetOptimizerTimeout());
-
-    if (query_type == network::QueryType::QUERY_DROP_INDEX) {
-      // Drop index does not need execution of compiled query
-      execution::sql::DDLExecutors::DropIndexExecutor(common::ManagedPointer<planner::AbstractPlanNode>(out_plan)
-                                                          .CastManagedPointerTo<planner::DropIndexPlanNode>(),
-                                                      common::ManagedPointer<catalog::CatalogAccessor>(accessor));
-    } else {
-      if (query_type == network::QueryType::QUERY_CREATE_INDEX) {
-        // TODO(lin): We actually don't need to populate the index tuples after creating the index placeholder for the
-        //  "what-if" API. But since we need to execute the query to get the features, we need to compile and execute
-        //  the query for now.
-        execution::sql::DDLExecutors::CreateIndexExecutor(common::ManagedPointer<planner::AbstractPlanNode>(out_plan)
-                                                              .CastManagedPointerTo<planner::CreateIndexPlanNode>(),
-                                                          common::ManagedPointer<catalog::CatalogAccessor>(accessor));
-      }
-
-      auto exec_query = execution::compiler::CompilationContext::Compile(*out_plan, exec_settings, accessor.get(),
-                                                                         execution::compiler::CompilationMode::OneShot);
-
-      auto exec_ctx = std::make_unique<execution::exec::ExecutionContext>(
-          db_oid, common::ManagedPointer(txn), callback, out_plan->GetOutputSchema().Get(),
-          common::ManagedPointer(accessor), exec_settings, pilot->metrics_thread_->GetMetricsManager());
-
-      exec_query->Run(common::ManagedPointer(exec_ctx), execution::vm::ExecutionMode::Interpret);
-    }
-    txn_manager->Commit(txn, transaction::TransactionUtil::EmptyCallback, nullptr);
+    bool success = true;
+    size_t idx = util.CompileQuery(sql_query, nullptr, nullptr, &success);
+    if (success) util.ExecuteQuery(idx, nullptr, nullptr, nullptr);
+    std::this_thread::sleep_for(std::chrono::seconds(1));
```

Comment:
Sleep?

Reply:
Then the merge conflict was resolved strangely.